yt_dlp/extractor/premiershiprugby.py

```diff
@@ -0,0 +1,37 @@
+from .common import InfoExtractor
+from ..utils import int_or_none, traverse_obj
+
+
+class PremiershipRugbyIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:\w+\.)premiershiprugby\.(?:com)/watch/(?P<id>[\w-]+)'
+    _TESTS = [{
+        'url': 'https://www.premiershiprugby.com/watch/full-match-harlequins-v-newcastle-falcons',
+        'info_dict': {
+            'id': '0_mbkb7ldt',
+            'title': 'Full Match: Harlequins v Newcastle Falcons',
+            'ext': 'mp4',
+            'thumbnail': 'https://open.http.mp.streamamg.com/p/3000914/sp/300091400/thumbnail/entry_id/0_mbkb7ldt//width/960/height/540/type/1/quality/75',
+            'duration': 6093.0,
+            'tags': ['video'],
+        }
+    }]
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        json_data = self._download_json(
+            f'https://article-cms-api.incrowdsports.com/v2/articles/slug/{display_id}?clientId=PRL',
+            display_id)['data']['article']
+
+        formats, subs = self._extract_m3u8_formats_and_subtitles(
+            json_data['heroMedia']['content']['videoLink'], display_id
+        )
+
+        return {
+            'id': json_data['heroMedia']['content']['sourceSystemId'],
+            'title': traverse_obj(json_data, ('heroMedia', 'title')),
+            'formats': formats,
+            'subtitles': subs,
+            'thumbnail': traverse_obj(json_data, ('heroMedia', 'content', 'videoThumbnail')),
+            'duration': int_or_none(traverse_obj(json_data, ('heroMedia', 'content', 'metadata', 'msDuration'))) / 1000,
```

Comment:
Currently, i don't know how to make this line safe if the `traverse_obj` return `None`. Is there any helper function that can make this safe ?

Reply:
Fixed